src/validation_utils.py

```diff
@@ -13,32 +18,69 @@ def type_is_equal(col_type: str, val_type: str) -> bool:
         return True
     return False
 
-def check_column_types(data, validation):
-    """Iterate over all csv files and check whether columns are well formatted."""
+def check_column_types(data: dict, validation: dict) -> list:
+    """Iterate over all csv files and check whether columns are well formatted.
+
+    Parameters
+    ----------
+    data:
+        Dictionary mapping from filename to Pandas.DataFrame
+    validation:
+        Dictionary rendered from the validation toml.
+
+    Returns
+    -------
+    List of tuples [(data_name, column_name, found_type, expected_type)]
+
+    """
+    output = []
+    
     for data_name in data:
         if data_name == "host_events.csv":
             section = "events"
         else:
             section = data_name.split(".")[0]
-        print(f"Checking {data_name} against {section}:")
         for var_name in validation[section]:
             if var_name in data[data_name].columns:
-                if not type_is_equal(str(data[data_name][var_name].dtype), validation[section][var_name]["type"]):
-                    print(f"{var_name} should be {validation[section][var_name]['type']}")
-                    print(f"\t Found: {str(data[data_name][var_name].dtype)}")
-    print("----")
+                if isinstance(validation[section][var_name]["type"], list):
```

Comment:
Some intermediate variables would make this code a lot more legible. Also, in this case the logic would be simpler if you first convert both cases to a list and then put the logic only on the list type.